File: 251002/Samoilenko/publisher/src/main/java/com/example/discussion/model/Writer.java
package com.example.discussion.model;

import jakarta.persistence.*;
import jakarta.validation.constraints.Size;

@Entity
@Table(name = "tbl_writer")
public class Writer {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(name = "login", nullable = false, length = 64)
    @Size(min = 2, max = 64, message = "Login must be between 2 and 64 characters")
    private String login;

    @Column(name = "password", nullable = false, length = 128)
    private String password;

    @Column(name = "firstname", nullable = false, length = 64)
    private String firstname;

    @Column(name = "lastname", nullable = false, length = 64)
    private String lastname;

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public String getLogin() {
        return login;
    }

    public void setLogin(String login) {
        this.login = login;
    }

    public String getPassword() {
        return password;
    }

    public void setPassword(String password) {
        this.password = password;
    }

    public String getFirstname() {
        return firstname;
    }

    public void setFirstname(String firstname) {
        this.firstname = firstname;
    }

    public String getLastname() {
        return lastname;
    }

    public void setLastname(String lastname) {
        this.lastname = lastname;
    }
}

File: 251003/Pankratiev/rest/publisher/src/main/kotlin/bsuir/dc/publisher/entity/Writer.kt
package bsuir.dc.publisher.entity

import jakarta.persistence.*

@Entity
@Table(name = "tbl_writer", schema = "public")
data class Writer(
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    var id: Long = 0,

    @Column(nullable = false, unique = true, length = 64)
    val login: String = "",

    @Column(nullable = false, length = 128)
    val password: String = "",

    @Column(nullable = false, length = 64)
    val firstname: String = "",

    @Column(nullable = false, length = 64)
    val lastname: String = "",

    @OneToMany(mappedBy = "writer", cascade = [CascadeType.ALL], orphanRemoval = true)
    val issues: MutableList<Issue> = mutableListOf()
)
